documentation/cadl-sample-project/specification/contosowidgetmanager/Contoso.WidgetManager.Shared/main.cadl

```diff
@@ -0,0 +1,8 @@
+@doc("Custom etag")
```

Comment:
If there is only a change in this common file, sdk automation pipeline cannot detect to generate SDK, which is smiliar to common files in swagger, and we cannot detect which sdk needs to be regenerated.

Reply:
This shared library is more likely used by this service, and SDK automation should be able identify the package under this service folder for generation.

Reply:
Will the shared library's name always endwith `Shared`? If yes, I think we can have the enhancement: when the shared library is changed, we will regenerate all the packages.
Moreoever, I think this feature is also needed in powerapps because powerapps doesn't support multi packages in one spec PR now in current scope.

Reply:
Yes, that is the convention - shared library folder always ends with `Shared`.

Reply:
There is a standard definition of ETag in ARM. I suggest `...` spread into your custom model.

Reply:
also since it is a sample project, I won't want folks to start down customization of standard models path but rather grow organically to add as needed/appropriately. So I think we can put some dummy shared/common models with instructions that if no teamwide shared/common artifacts, this project can be removed.

Reply:
Updated to use a faked shared model.